src/azure-cli/azure/cli/command_modules/appservice/tunnel.py

```diff
@@ -91,23 +92,32 @@ def is_webapp_up(self):
         if should_disable_connection_verify():
             http = urllib3.PoolManager(cert_reqs='CERT_NONE')
         headers = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(self.remote_user_name, self.remote_password))
-        url = 'https://{}{}'.format(self.remote_addr, '/AppServiceTunnel/Tunnel.ashx?GetStatus')
+        url = 'https://{}{}'.format(self.remote_addr, '/AppServiceTunnel/Tunnel.ashx?GetStatus&GetStatusAPIVer=2')
         r = http.request(
             'GET',
             url,
             headers=headers,
             preload_content=False
         )
+
+        logger.warning('Verifying if app is running....')
+
         if r.status != 200:
             raise CLIError("Failed to connect to '{}' with status code '{}' and reason '{}'".format(
                 url, r.status, r.reason))
-        msg = r.read().decode('utf-8')
-        logger.info('Status response message: %s', msg)
-        if 'FAIL' in msg.upper():
-            logger.info('WARNING - Remote debugging may not be setup properly. Reponse content: %s', msg)
-            return False
-        if 'SUCCESS' in msg.upper():
-            return True
+        resp_msg = r.read().decode('utf-8')
+        json_data = json.loads(resp_msg)
+
+        if 'STARTED' in json_data["state"].upper():
```

Comment:
Is it possible the key is not found

Reply:
Added a check if key is not found, returning false from this method would invoke the retry logic